### DISCUSSION TITLE
The easy MVC way to get content without defining model? {% assign services = Content["slug:services"] %}

### DISCUSSION THREAD

**Discussion by orchardc** [OrchardCMS/OrchardCore#8022] [General]
In Liquid page, it is fairly easy to get the content and it's id, and then query the BagPart without the need to define any Model - as code shown below.

Is there a similar easy method in MVC?? Currently the only way I know is to query the database and use JSONParse to Models.

{% assign services = Content["slug:services"] %}
{% assign serviceContentItemId = Request.Path | split: "/" | last | split: "-" | first %}

{% for service in services.Content.BagPart.ContentItems %}
//You can work on contents ..........
{% endfor %}

**Comment by Skrypt** [comment]
You need to make use of `jsonData.As<EntityName>();` which requires a reference from OrchardCore.Entities

**Comment by orchardc** [comment]
Sorry @Skrypt , I don't get it.  Are you suggesting you can use it the MVC view directly without defining model or using controller to get the object first?

Do you have example? Thank you!

**Reply by Skrypt** [reply]
I assume you meant using a Razor view because you said "MVC". And, yes you can do this directly in the view if you use the appropriate `@using Something`. Though, the proper way to do this is to pass it with you ViewModel of course.

**Reply by orchardc** [reply]
![image](https://user-images.githubusercontent.com/16193516/102725142-61897900-430c-11eb-8033-7c04c1181246.png)

It's for MVC view, yes. But there is an error when using it...

**Reply by hishamco** [reply]
If I'm not wrong @Skrypt means to use your JSON object, then cast it using `As<TModel>`

**Reply by Skrypt** [reply]
hmm you need to declare your variable first?
Also, here, where does your MyProduct comes from? Is it a Content Type or Content Part, is it a model from a custom service?
The remaining of the solution for you depends on these details.

If I presume from your example : `services.Content.BagPart.ContentItems`
These are bags content items.
Then you should be able to do quite the same with

```C#
@using OrchardCore.Entities;
@using YourAssembly.WhereYourMyProductIs;
@
{
    var contentItems = Model.Content.BagPart.ContentItems;
    foreach(contentItem in contentItems)
    {
        var myProduct = contentItem.As<MyProduct>();
    }
}
```

**Reply by orchardc** [reply]
MyProduct is a Content Type, which contains a Content Part: Bag Part which links to other part.

I'm not sure if it's possible to get the object without defining a Model - just like the Liquid page does (if you see the example I listed above)?

**Reply by orchardc** [reply]
@hishamco ,thanks for your response.

Say I can get the json data of the Content of the MyProduct in a controller:

```
 var jsonData = JObject.Parse(myProduct.Content);
 var model = jsonData.As<dynamic>();
```

Then how to access the dynamic property and its values from the model object?

**Reply by Skrypt** [reply]
Here, the Content Type is represented in a shape which is a dynamic object (ExpandoObject). 
Normally we don't cast. We use it as a dynamic object.

`Model.Content.ContentPartName?.FieldName` with null conditional operator to validate it has a value.
I rarely cast these objects back to strongly typed objects for the Views.
